File: src/sdk/rust/src/server_wss.rs
// server_wss.rs
// Server SDK implementation for App Mesh (WSS)

use std::sync::Arc;

use crate::client_wss::AppMeshClientWSS;
use crate::error::AppMeshError;
use crate::server_http::AppMeshWorker;

/// Server-side helper for applications using WSS transport
pub struct AppMeshWorkerWSS {
    server: Arc<AppMeshWorker>,
}

impl AppMeshWorkerWSS {
    pub fn new(
        wss_address: Option<(String, u16)>,
        ssl_verify: Option<String>,
        ssl_client_cert: Option<(String, String)>,
    ) -> Result<Arc<Self>, AppMeshError> {
        let client = AppMeshClientWSS::new(wss_address, ssl_verify, ssl_client_cert)?;
        // Server endpoints use APP_MESH_PROCESS_KEY; no JWT refresh needed.
        client.client().set_auto_refresh_token(false);
        Ok(Arc::new(Self { server: AppMeshWorker::with_client(client.client().clone()) }))
    }
}

impl std::ops::Deref for AppMeshWorkerWSS {
    type Target = AppMeshWorker;

    fn deref(&self) -> &Self::Target {
        &self.server
    }
}

File: src/sdk/go/server_wss.go
package appmesh

// AppMeshServerWssContext provides an App Mesh server interface that communicates over a WSS transport
type AppMeshServerWssContext struct {
	AppMeshServerHttpContext
}

// NewWSSContext creates a server-side task context over WSS. Server endpoints
// authenticate via APP_MESH_PROCESS_KEY, not JWT, so token refresh is forced off.
func NewWSSContext(options Option) (*AppMeshServerWssContext, error) {
	options.AutoRefreshToken = false
	wssClient, err := NewWSSClient(options)
	if err != nil {
		return nil, err
	}

	server, err := newHTTPContextWithRequester(options, wssClient.wssReq)
	if err != nil {
		return nil, err
	}

	return &AppMeshServerWssContext{AppMeshServerHttpContext: *server}, nil
}

// CloseConnection closes the underlying WSS connection used by this server context.
func (r *AppMeshServerWssContext) CloseConnection() {
	r.client.req.Close()
}
